cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -358,14 +372,50 @@ installAction (configFlags, configExFlags, installFlags, haddockFlags, newInstal
           | Just (pkg :: PackageId) <- simpleParse pkgName = return pkg
           | otherwise = die' verbosity ("Invalid package ID: " ++ pkgName)
       packageIds <- mapM parsePkg targetStrings
+      
+      cabalDir <- getCabalDir
```

Comment:
I'd make this whole bit a separate function with a name like `reportCasingErrors` or `checkForUnknownPackages`.